### ISSUE TITLE
How to properly end session? "Enhance pointer precision" in PC mouse settings always stays on after a session

### ISSUE THREAD

**Issue by philhzss** [FrogTheFrog/moondeck-buddy#18]
Hi,

Just setup this little thing, love it!! Great job. I just have one small issue, I never use mouse acceleration on my computer, it's always off. However, I've noticed that after a gaming session over Moonlight launched with this plugin, this option is on:
![image](https://user-images.githubusercontent.com/41757256/211166336-0962b683-53b5-4351-99b8-cb938ecb44aa.png)

I just did some testing using Moonlight directly (not your plugin), and it seems that Moonlight does also enable mouse acceleration, but when I end the stream / close the game, the acceleration gets turned back off.

Therefor, I assume I'm not ending the session properly when using your plugin. I've tried both closing the game (which appears to close everything properly, and then the Steam Deck returns to the library), and I've also tried force closing a session using the right hand menu, mouse acceleration stays on.

Do you know what I'm doing wrong? Is there an additional step I need to do to properly end a streaming session?

Thanks a lot!!!

**Comment by FrogTheFrog** [comment]
Can you post your moonlight settings? I can not get the precision enabled for me no matter what I do...

Are you using Sunshine (if yes, which version) or Nvidia Gamestream?

**Comment by philhzss** [comment]
Here are my moonlight settings on the deck. I'm using NVIDIA gamestream for now, using your latest Buddy release that re-adds support for it.

![20230107_144745.jpg](https://user-images.githubusercontent.com/41757256/211168173-ca845b3a-de56-4e40-899a-bbfdd6ca725f.jpg)
![20230107_144814.jpg](https://user-images.githubusercontent.com/41757256/211168179-5c466ccf-9198-4d77-8455-02f0baf6fefb.jpg)
![20230107_144834.jpg](https://user-images.githubusercontent.com/41757256/211168181-1fa158c8-33bb-4bfe-823f-413a72e22804.jpg)

I just remembered I have [gamestream launchpad](https://github.com/cgarst/gamestream_launchpad) on my PC, not sure if it has anything to do with my problem but I will look into it right now! 

Thanks

**Comment by FrogTheFrog** [comment]
Don't worry I have replicated it. Will post explanation in a few minutes...

**Comment by FrogTheFrog** [comment]
@philhzss This is the issue summarized with Nvidia:
![image](https://user-images.githubusercontent.com/22381748/211168550-86c8d6de-91c1-4932-bbec-b321759393d1.png)

Since I'm hacking around for Nvidia, I will check if it would be hard to reset the previous state

**Comment by FrogTheFrog** [comment]
Nvidia does not just does not care about any custom process and you have to manually kill the `nvstreamer.exe` to terminate the stream...

**Comment by philhzss** [comment]
Ah okay I see! Thanks for finding out why. If you didn't have the issue initially (when you couldn't replicate it), what did you do to be able to replicate it, maybe I can just do what you did?

Hope the fix is easy for you, but if it's a pain then it might not be worth it!! Thanks for looking into it so quickly though

**Comment by FrogTheFrog** [comment]
I'm using Sunshine already as the performance is 1:1 for me on nightly (had to set monitor refresh rate to above 60Hz for some weird reason).

**Comment by philhzss** [comment]
Oh really. I might try it then .. I have never tried sunshine, I had read it's not as good as gamestream so I wanted to keep using gamestream as long as I can, even if it's a dead product, but I might give sunshine I try now!!

**Comment by FrogTheFrog** [comment]
@philhzss Please try the 1.3.2. Launch the Buddy, close it, open `settings.json` where Buddy is located, change one obvious value to `true` and now Buddy should always disable mouse acceleration after `nvstreamer.exe` is killed.

**Comment by philhzss** [comment]
@FrogTheFrog It works!! Perfect, thank you so much for doing this so quickly!! You can close this

Side note, the readme says `Additionally, it allows to: [...] close the Steam process;`, is this a setting somewhere? I would like it to keep Steam open, but everytime I stop playing Steam gets shutdown on my main PC.

**Comment by FrogTheFrog** [comment]
> Side note, the readme says `Additionally, it allows to: [...] close the Steam process;`, is this a setting somewhere? I would like it to keep Steam open, but everytime I stop playing Steam gets shutdown on my main PC.

I needed to do it because of Nvidia. I will see it in the future if this can be avoided.

**Comment by philhzss** [comment]
> > Side note, the readme says `Additionally, it allows to: [...] close the Steam process;`, is this a setting somewhere? I would like it to keep Steam open, but everytime I stop playing Steam gets shutdown on my main PC.
> 
> I needed to do it because of Nvidia. I will see it in the future if this can be avoided.

Ok cool thank you!! I had read it as "it can close the process" I wasn't sure if you meant it has to or it's a choice. Thanks!!

**Comment by FrogTheFrog** [comment]
In the upcoming release the Steam will no longer be closed by default, but will rain open in big picture mode (because there's no way to exit it via command line)

**Comment by philhzss** [comment]
Ok sounds good!! Yeah I just asked because I had to use my phone and remote desktop to manually reopen Steam after each session, because if I didn't then I couldn't launch another session from the Deck. Sounds good

**Comment by FrogTheFrog** [comment]
Hmm, the Buddy is able to manually launch Steam whenever it wants to launch game (at least for me). But there could be some permission issues that I don't know about...

**Comment by philhzss** [comment]
Oh okay, wait a second I'll do a few tests

**Comment by philhzss** [comment]
No you're right, I just tested and the games do launch, Buddy reopens Steam... that didn't work yesterday, I was getting an error on the Deck, when after opening Steam manually I thought it had fixed the error. But now it works even without opening Steam, so I must have had something set wrong!

> Hmm, the Buddy is able to manually launch Steam whenever it wants to launch game (at least for me). But there could be some permission issues that I don't know about...

And it's like the audio, I was about to ask you if you know why sometimes there is no audio (I know you talk about having another "sink" in the readme, I actually do use VB-CABLE Virtual Audio Device and Voicemeeter since a long time... but now with my tests, it seems that only a specific game is having isssues?? Teardown. I tested other games and I have audio. In fact, weirdly if I minimize Teardown and play an audio file on my PC during a session, I have audio... it's just the game itself that has no sound. And even more weirdly, if I start the session using the Moonlight app on my Deck instead of using Moondeck and buddy, I DO have audio in Teardown... in both cases my PC's audio output is automatically set to "NVIDIA HDMI Out" (still using NVIDIA for now)

So yeah, further testing shows that everything does seem to work like it should, I'm just having minor problems with specific situations

**Comment by FrogTheFrog** [comment]
The audio is because of Nvidia and the "I will launch the app, but pretend I don't know about it issue".

The reason why MoonDeckStream was added in the first place is to work around the instability of Nvidia's `Steam` app that it uses. It just breaks in various ways with the new big picture mode from beta branch.

But then the news hit and I noticed that the initial solution works for Sunshine without any issues, so I just dropped the support for Nvidia initially. But now people are using it and Nvidia's bugs keep popping up...

**Comment by philhzss** [comment]
Well, considering that NVIDIA is killing Gamestream anyway, you should maybe not work on all these nvidia problems?? I haven't bothered to switch to Sunshine yet because I've had Gamestream on my PC for a while and it worked, but sunshine supports more systems and is not going to be killed in 1 month, so perhaps its not worth it for you to try to fix all these NVIDIA problems???
I'll give sunshine a try soon!!

**Comment by FrogTheFrog** [comment]
Just use the Sunshine's nightly release!

I also had this issue, but it seems not everyone has it (copy pasta):

So I have this issue that I've managed to solve on Windows (Nvidia). I'm streaming games with a refresh rate set to 60Hz on the host, vsync enabled. On GFE everything is butter smooth, on Sunshine not so much. In addition I also have a visible stutter of some sorts every 5 seconds. Disabled the vsync - did not help.

I compared the Moonlight FPS reading from when running GFE and Sunshine. With GFE the FPS range is 59 < fps < 61. On Sunshine it's 59 < fps < 60. So I went on an increased refresh rate on the host's monitor to 120Hz and it suddenly became butter smooth and the stutter was gone (FPS range also became 59 < fps < 61).

I then went back on to enable vsync - still the same result, all is good.

I went a step further and enabled the FPS limit on Nvidia control panel to 60 FPS (fps range remained the same). Still the same result, butter smooth, no stutter and saving electricity :).

I went around and played more with the refresh rate. It seems that any refresh rate that was above 60Hz (85, 100, 120, 144) removed the stutter and the smoothness was very comparable. If it felt different I think it's most likely a placebo at this point.

And so here I am with a question, is this a bug or a feature? Should I open an issue?

Basically everyone that use that cheap HDMI fake display dongle with up to 60Hz is f***ed otherwise...
(unless my PC is a snowflake and I'm the only one affected :/)
